cartridges/int_bolt_sfra/cartridge/client/default/js/boltProductPageButton.js

```diff
@@ -1,22 +1,42 @@
 'use strict';
 
-// initProductPageCheckoutButton Polls:
+// showBoltProductPageCheckoutButton Polls:
 // 1) the DOM for the product page checkout button container
 // 2) BoltConnect
 // when they become available, it attaches an onclick event handler to it.
-var initProductPageCheckoutButton = function () {
+var showBoltProductPageCheckoutButton = function () {
     var boltButtonExist = setInterval(function () {
-        var checkoutBoltButton = $('.bolt-product-checkout-button')[0]; // @ts-ignore
+        var checkoutBoltButton = $('.bolt-button-wrapper[data-tid="instant-bolt-checkout-button"]'); // @ts-ignore
         // have to check if child of this dom is svg, otherwise bolt button is not fully rendered (it's the object)
-        var boltButtonLoaded = checkoutBoltButton && window.BoltCheckout;
+        var boltButtonLoaded = checkoutBoltButton && window.BoltCheckout && checkoutBoltButton.children()[0].nodeName === 'svg';
         if (boltButtonLoaded) {
             clearInterval(boltButtonExist);
             console.log('Bolt connect has loaded and product page button is ready!');
         }
     }, 250);
 };
 
+var callBoltConfigure = function (product) {
+    if (product.available && product.readyToOrder) {
+        console.log('Ready to call BoltCheckout.Configure');
```

Comment:
@var-bolt Is this for debugging only? I do not see any logic in this function.

Reply:
I have since updated the PR! Please take a second look 😄